File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/cases/HttpIntegrationCases.java
package org.galaxio.performance.picatinny.cases;

import io.gatling.javaapi.core.ChainBuilder;

import static io.gatling.javaapi.core.CoreDsl.exec;
import static io.gatling.javaapi.http.HttpDsl.*;

public final class HttpIntegrationCases {
    private HttpIntegrationCases() {}

    public static ChainBuilder echo() {
        return exec(
            http("echo")
                .get("/echo/#{ts}")
                .header("Authorization", "Bearer #{jwt}")
                .check(status().is(200))
                .check(jsonPath("$.ts").is("#{ts}"))
                .check(jsonPath("$.ts").<Boolean>transform(v -> v.matches("\\d{17}")).is(Boolean.TRUE))
                .check(jsonPath("$.auth").is("Bearer #{jwt}"))
                .check(jsonPath("$.auth").<Boolean>transform(
                    v -> v.matches("Bearer [\\w-]+\\.[\\w-]+\\.[\\w-]+")).is(Boolean.TRUE))
        );
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/cases/HttpIntegrationCases.kt
package org.galaxio.performance.picatinny.cases

import io.gatling.javaapi.core.ChainBuilder
import io.gatling.javaapi.core.CoreDsl.exec

import io.gatling.javaapi.http.HttpDsl.*

object HttpIntegrationCases {
    fun echo(): ChainBuilder = exec(
        http("echo")
            .get("/echo/#{ts}")
            .header("Authorization", "Bearer #{jwt}")
            .check(status().shouldBe(200))
            .check(jsonPath("$.ts").shouldBe("#{ts}"))
            .check(jsonPath("$.ts").transform { it.matches(Regex("\\d{17}")) }.shouldBe(true))
            .check(jsonPath("$.auth").shouldBe("Bearer #{jwt}"))
            .check(jsonPath("$.auth").transform { it.matches(Regex("Bearer [\\w-]+\\.[\\w-]+\\.[\\w-]+")) }.shouldBe(true))
    )
}
